api/apps/api/src/modules/clone/import/domain/import/import.ts

```diff
@@ -16,28 +17,49 @@ import { ImportSnapshot } from './import.snapshot';
 
 export const componentNotFound = Symbol(`component not found`);
 export const componentAlreadyCompleted = Symbol(`component already completed`);
+export const componentAlreadyFailed = Symbol(`component already failed`);
 
 export type CompletePieceSuccess = true;
 export type CompletePieceErrors =
   | typeof componentNotFound
   | typeof componentAlreadyCompleted;
+export type MarkPieceAsFailedErrors =
+  | typeof componentNotFound
+  | typeof componentAlreadyFailed;
 
 export class Import extends AggregateRoot {
   private constructor(
     readonly importId: ImportId,
     private readonly resourceId: ResourceId,
     private readonly resourceKind: ResourceKind,
+    private readonly projectId: ResourceId,
     private readonly archiveLocation: ArchiveLocation,
     private readonly pieces: ImportComponent[],
   ) {
     super();
+
+    const isProjectImport = resourceKind === ResourceKind.Project;
+    const equalResourceAndProjectId = resourceId.equals(projectId);
+
+    if (isProjectImport && !equalResourceAndProjectId) {
```

Comment:
I think a better naming could be `resourceIdAndProjectIdAreEqual`, as it is more semantic